genie-client/src/main/java/com/netflix/genie/client/interceptors/ResponseMappingInterceptor.java

```diff
@@ -53,17 +56,24 @@ public Response intercept(final Chain chain) throws IOException {
         if (response.isSuccessful()) {
             return response;
         } else {
-            final String bodyContent = response.body().string();
+            final ResponseBody body = response.body();
+            if (body != null) {
+                final Reader bodyReader = body.charStream();
 
-            try {
-                final JsonNode responseBody = this.mapper.readTree(bodyContent);
-                throw new GenieClientException(
-                    response.code(),
-                    response.message() + " : " + responseBody.get("message")
-                );
-            } catch (final JsonProcessingException jpe) {
-                throw new GenieClientException(response.code(), response.message() + bodyContent);
+                if (bodyReader != null) {
+                    try {
+                        final JsonNode responseBody = this.mapper.readTree(bodyReader);
```

Comment:
Shouldn't it check for the possibility that `responseBody` is null?

Reply:
Yeah. The documentation on the `readTree` method was a little vague but better safe than sorry. Will fix.